coops-ui/src/views/StandaloneDirectorsFiling.vue

```diff
@@ -0,0 +1,301 @@
+import { Affix } from 'vue-affix'
+import SbcFeeSummary from 'sbc-common-components/src/components/SbcFeeSummary.vue'
+import { mapState, mapActions, mapGetters } from 'vuex'
+
+export default {
+  name: 'StandaloneDirectorsFiling',
+
+  components: {
+    Directors,
+    SbcFeeSummary,
+    Affix
+  },
+
+  data () {
+    return {
+      showLoading: false,
+      loadingMsg: 'Redirecting to PayBC to Process Your Payment',
+      directorsChange: false,
+      filingData: [],
+      dialog: false,
+      lastFilingDate: 'something' // TODO
+    }
+  },
+
+  computed: {
+    ...mapState(['noAGM', 'regOffAddrChange',
+      'validated', 'currentDate', 'corpNum',
+      'entityName', 'entityIncNo', 'entityFoundingDate', 'currentARStatus',
+      'addressesFormValid', 'directorFormValid'])
+
+  },
+
+  mounted () {
+    // if tombstone data isn't set, redirect to home
+    if (!this.corpNum) {
+      this.$router.push('/')
+    } else {
+      // load initial data
+      // TODO - anything here?
+    }
+  },
+
+  methods: {
+    ...mapActions(['setDirectorFormValid', 'setValidated']),
+
+    directorsChangeEventHandler (val) {
+      this.directorsChange = val
+    },
+
+    submit () {
+      let changeOfDirectors = null
+
+      const header = {
+        header: { name: 'changeOfDirectors', date: this.currentDate }
+      }
+      const business = {
+        business: {
+          foundingDate: this.entityFoundingDate,
+          identifier: this.entityIncNo,
+          legalName: this.entityName
+        }
+      }
+
+      if (this.isDataChanged('OTCDR')) {
+        changeOfDirectors = {
+          changeOfDirectors: {
+            certifiedBy: 'Full Name',
+            email: 'no_one@never.get',
+            directors: this.$refs.directorsList.getAllDirectors()
+          }
+        }
+      }
+
+      const filingData = {
+        filing: Object.assign(
+          {},
+          header,
+          business,
+          changeOfDirectors
+        )
+      }
+
+      // Temp fix Added to remove cors error while invoking mock endpoint.
+      // To be removed when the API is in place
+      let config = {
+        headers: {
+          'Content-Type': 'text/plain'
+        }
+      }
+
+      axios.post(this.corpNum + '/filings', filingData, config).then(res => {
+        let payRequestId:String = res.data.filing.header.paymentToken
+        payRequestId = '189'
+        let returnURL = window.location.origin + '/standalone-directors?pay_id=' + payRequestId
+        let authStub:string = this.authURL
+        if (!(authStub.endsWith('/'))) {
+          authStub = authStub + '/'
+        }
+        let payURL = authStub + 'makepayment/' + payRequestId + '/' + encodeURIComponent(returnURL)
+        window.location.href = payURL
+      }).catch((error) => {
+        // TODO : To Decide how and where to display the error message from API
+        console.log(error)
+        this.dialog = true
+      })
+    },
+
+    toggleFiling (setting, filing) {
```

Comment:
Something for us to eventually figure out: this (and other code) is duplicated from AnnualReport.vue. Can we have this kind of common code in a base TypeScript class (or Component? does that work?).

Reply:
I might (by the end of this implementation) be able to break this out into a generic "Filing" component or class. For now you are right, there's a lot copied and I don't love it either but time is crunched. I'm also adding another standalone after this batch is approved, for standalone change of address, so it will be worse :)